Pidaq/Lib/can_manager/can_manager.py

```diff
@@ -0,0 +1,169 @@
+"""Contains classes and methods to implement and manage the can (socketcan) bus
+
+Classes:
+    CanManager
+    SensorReading
+
+Functions:
+    parse_conversion_factor
+"""
+import json
+import os
+
+import can
+
+
+class CanManager:
+    """Manages the can bus, and sets the SensorReading objects for the test being run
+
+    This class contains the socketcan bus instance that will be used to communicate with
+    the sensor board, as well as the configured sensor readings for the particular project
+    that is being tested
+
+    Attributes:
+        bus (can.interfaces.socketcan.SocketcanBus)
+        messages (dict): dict of all the messages to be expected from the 
+                         sensor board. Keys are message ids, and values are
+                         SensorReading objects
+        message_ids (list(int)): list of all the message ids, hexidecimal integers
+
+    Methods:
+        read_message_config(project: str)
+            Reads the configuration of sensor readings from messageconfig.json
+        send_message(id: int, data: list)
+        read_bus
+            Reads the bus for a message
+        assign_message_data(bus_message: can.Message)
+            assigns data to the correct SensorReading object
+    """
+    def __init__(self, bus_name: str) -> None:
+        self.bus = can.interfaces.socketcan.SocketcanBus(channel=bus_name)
+        self.messages = {}
+        self.message_ids = []
+
+    def read_message_config(self, project: str, config_file: str, path=None) -> None:
+        """Reads sensor readings configuration from messageconfig.json
+
+        Constructs SensorReading objects for all expected sensor readings,
+        and appends them to a list
+
+        Parameters:
+            project (str): current project, must be one of dts, suspension, or windtunnel
+            config_file (str): message configuration file name
+            path (str): path to configuration file, if no path specified, current working
+                        directory is assumed
```

Comment:
I cant remember if I previously commented this or not - gonna add it on this review as well though. This path should be described as the path to the folder containing the config file.